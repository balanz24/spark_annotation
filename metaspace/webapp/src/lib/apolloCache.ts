import { defaultDataIdFromObject, InMemoryCache } from 'apollo-client-preset'

const nonNormalizableTypes: any[] = ['User', 'DatasetUser', 'DatasetGroup', 'DatasetProject']
export const makeApolloCache = () => new InMemoryCache({
  cacheRedirects: {
    Query: {
      // Allow get-by-id queries to use cached data that originated from other kinds of queries
      dataset: (_, args, { getCacheKey }) => getCacheKey({ __typename: 'Dataset', id: args.id }),
      annotation: (_, args, { getCacheKey }) => getCacheKey({ __typename: 'Annotation', id: args.id }),
      group: (_, args, { getCacheKey }) => getCacheKey({ __typename: 'Group', id: args.groupId }),
      project: (_, args, { getCacheKey }) => getCacheKey({ __typename: 'Project', id: args.projectId }),
    },
  },
  dataIdFromObject(object: any) {
    // WORKAROUND: Because of Apollo's aggressive caching, often the current User will be overwritten with results
    // from other queries. The server side often strips fields based on how they're accessed (the "ScopeRole" logic),
    // which means these query paths will often return different data with the same IDs:
    // currentUser -> primaryGroup (always present)
    // dataset -> submitter -> primaryGroup (null unless admin)
    // To protect against this, don't allow Users (and possibly other types in the future) to have a dataId,
    // so that InMemoryCache cannot share data between different queries.
    if (nonNormalizableTypes.includes(object.__typename)) {
      return null
    } else {
      return defaultDataIdFromObject(object)
    }
  },
})
