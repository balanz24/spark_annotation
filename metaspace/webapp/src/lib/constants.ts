export const MAX_MOL_DBS = 4
export const MAX_MOL_DBS_EXT = 6
export const MAX_NEUTRAL_LOSSES = 3
export const MAX_CHEM_MODS = 3
export const DEFAULT_SCALE_TYPE = 'linear'
