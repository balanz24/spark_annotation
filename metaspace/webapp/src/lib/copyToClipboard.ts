export default function copyToClipboard(text?: string) {
  if (text) {
    if ('clipboard' in navigator) {
      navigator.clipboard.writeText(text)
    } else {
      const el = document.createElement('textarea')
      el.value = text
      el.style.position = 'absolute'
      el.style.left = '-9999px'
      document.body.appendChild(el)
      try {
        el.select()
        document.execCommand('copy')
      } finally {
        document.body.removeChild(el)
      }
    }
  }
}
