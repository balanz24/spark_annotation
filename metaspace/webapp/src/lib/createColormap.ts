import * as d3 from 'd3'
import getColorScale from './getColorScale'

export type OpacityMode = 'constant' | 'linear' | 'quadratic';

const OPACITY_MAPPINGS: Record<OpacityMode, (x:number) => number> = {
  constant: (x) => 255,
  linear: (x) => x,
  quadratic: (x) => 255 - (255 - x) * (255 - x) / 255,
}

export default function createColormap(name: string, opacityMode: OpacityMode = 'constant',
  opacityScale: number = 1): number[][] {
  const { domain, range } = getColorScale(name)
  const sclFun = d3.scaleLinear<string>().domain(domain).range(range).clamp(true)
  const alphaFunc = OPACITY_MAPPINGS[opacityMode]

  const colors = []
  for (let i = 0; i < 256; i++) {
    const color = d3.rgb(sclFun(i / 255.0))
    const alpha = alphaFunc(i) * opacityScale
    colors.push([color.r, color.g, color.b, alpha].map(Math.round))
  }
  return colors
}
