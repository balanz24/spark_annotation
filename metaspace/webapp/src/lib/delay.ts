export default function delay(timeMs: number) {
  return new Promise(resolve => setTimeout(resolve, timeMs))
}
