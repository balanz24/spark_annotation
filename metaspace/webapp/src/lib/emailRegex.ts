// W3C's validation regex for <input type="email">. This is quite permissive, but doesn't allow unicode domain names
// (e.g. foo@mañana.com). To handle unicode domain names, the server-side code would have to be changed to automatically
// convert them to punycode when sending emails, as AWS SES doesn't support unicode domain names.

export default /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
