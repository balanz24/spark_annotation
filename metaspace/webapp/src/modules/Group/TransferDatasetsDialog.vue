<template>
  <el-dialog
    class="dialog"
    visible
    append-to-body
    :title="allDatasets.length > 0 ? 'Transfer datasets' : 'Join group'"
    @close="handleClose"
  >
    <div v-loading="loading">
      <div v-if="allDatasets.length > 0">
        <p style="margin-top: 0">
          You have previously uploaded one or more datasets without a group. Do you want to transfer any of these datasets to {{ groupName }}?
        </p>
        <dataset-checkbox-list
          v-model="selectedDatasets"
          :datasets="allDatasets"
          :init-select-all="true"
        />
        <p v-if="!isInvited">
          An email will be sent to the group's principal investigator to confirm your access.
        </p>
        <div class="button-bar">
          <el-button
            :disabled="isSubmitting"
            @click="handleClose"
          >
            Cancel
          </el-button>
          <el-button
            type="primary"
            :loading="isSubmitting"
            @click="handleAccept"
          >
            {{ acceptText }}
          </el-button>
        </div>
      </div>
      <div v-else>
        <p
          v-if="!isInvited"
          style="margin-top: 0"
        >
          An email will be sent to the group's principal investigator to confirm your access.
        </p>
        <p
          v-else
          style="margin-top: 0"
        >
          Are you sure you want to join {{ groupName }}?
        </p>
        <div class="button-bar">
          <el-button
            :disabled="isSubmitting"
            size="small"
            @click="handleClose"
          >
            Cancel
          </el-button>
          <el-button
            type="primary"
            :loading="isSubmitting"
            size="small"
            @click="handleAccept"
          >
            {{ acceptText }}
          </el-button>
        </div>
      </div>
    </div>
  </el-dialog>
</template>
<script lang="ts">
import Vue from 'vue'
import { Component, Prop } from 'vue-property-decorator'
import { DatasetListItem, datasetListItemsQuery } from '../../api/dataset'
import DatasetCheckboxList from '../../components/DatasetCheckboxList.vue'
import { currentUserIdQuery, CurrentUserIdResult } from '../../api/user'

  @Component<TransferDatasetsDialog>({
    components: {
      DatasetCheckboxList,
    },
    apollo: {
      currentUser: {
        query: currentUserIdQuery,
        loadingKey: 'loading',
        fetchPolicy: 'cache-first',
      },
      allDatasets: {
        query: datasetListItemsQuery,
        loadingKey: 'loading',
        variables() {
          return {
            dFilter: {
              submitter: this.currentUser!.id,
              hasGroup: false,
            },
          }
        },
        skip() {
          return !this.currentUser || !this.currentUser.id
        },
      },
    },
  })
export default class TransferDatasetsDialog extends Vue {
    @Prop({ required: true })
    groupName!: string;

    @Prop({ required: true })
    isInvited!: boolean; // True to show "Accept/reject invitation", false to show "Request access/cancel"

    loading: number = 0;
    currentUser: CurrentUserIdResult | null = null;
    allDatasets: DatasetListItem[] = [];
    selectedDatasets: Record<string, boolean> = {};
    isSubmitting: Boolean = false;

    get numSelected() {
      return Object.values(this.selectedDatasets).filter(selected => selected).length
    }

    get acceptText() {
      const action = this.isInvited ? 'Join group' : 'Request access'
      return this.numSelected === 0
        ? action
        : `${action} and transfer ${this.numSelected} ${this.numSelected === 1 ? 'dataset' : 'datasets'}`
    }

    handleClose() {
      if (!this.isSubmitting) {
        this.$emit('close')
      }
    }

    handleAccept() {
      const selectedDatasetIds = Object.keys(this.selectedDatasets).filter(key => this.selectedDatasets[key])
      this.$emit('accept', selectedDatasetIds)
      this.isSubmitting = true
    }
}
</script>
<style scoped lang="scss">
  .dialog /deep/ .el-dialog {
    @apply max-w-lg;
  }
  .button-bar {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-end;
    margin-top: 20px;
  }
</style>
