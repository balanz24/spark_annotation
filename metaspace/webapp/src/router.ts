/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DatasetOverviewPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/overview/DatasetOverviewPage'),
  DatasetComparisonPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/comparison/DatasetComparisonPage'),
  DatasetBrowserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/imzml/DatasetBrowserPage'),
  DatasetEnrichmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/enrichment/DatasetEnrichmentPage'),
  SpottingProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "SpottingProjectPage" */ './modules/SpottingProject/DashboardPage'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/GroupsListPage'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Group/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),

  // Separate bundle for design docs
  DesignTOC: () => import(/* webpackChunkName: "DesignBundle" */ './design/TOCPage.vue'),
  DesignStyleGuide: () => import(/* webpackChunkName: "DesignBundle" */ './design/StyleGuidePage.vue'),
  DesignIcons: () => import(/* webpackChunkName: "DesignBundle" */ './design/IconsPage.vue'),
  DesignComponents: () => import(/* webpackChunkName: "DesignBundle" */ './design/ComponentsPage.vue'),
  DesignForms: () => import(/* webpackChunkName: "DesignBundle" */ './design/FormsPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', name: 'annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    {
      path: '/datasets/:dataset_id/comparison',
      name: 'datasets-comparison',
      component: asyncPages.DatasetComparisonPage,
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/dataset/:dataset_id', name: 'dataset-overview', component: asyncPages.DatasetOverviewPage },
    { path: '/dataset/:dataset_id/annotations', name: 'dataset-annotations', component: asyncPages.AnnotationsPage },
    { path: '/dataset/:dataset_id/browser', name: 'dataset-browser', component: asyncPages.DatasetBrowserPage },
    { path: '/dataset/:dataset_id/enrichment', name: 'dataset-enrichment', component: asyncPages.DatasetEnrichmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/groups', component: asyncPages.GroupsListPage },
    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignTOC, meta: { footer: true, flex: true } },
    { path: '/design/styleguide', component: asyncPages.DesignStyleGuide, meta: { footer: true, flex: true } },
    { path: '/design/icons', component: asyncPages.DesignIcons, meta: { footer: true, flex: true } },
    { path: '/design/components', component: asyncPages.DesignComponents, meta: { footer: true, flex: true } },
    { path: '/design/forms', component: asyncPages.DesignForms, meta: { footer: true, flex: true } },

    { path: '/detectability', name: 'detectability', component: asyncPages.SpottingProjectPage },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
