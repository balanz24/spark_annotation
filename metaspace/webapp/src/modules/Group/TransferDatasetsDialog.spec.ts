import { mount, Stubs, Wrapper } from '@vue/test-utils'
import { Button, Checkbox } from '../../lib/element-ui'
import Vue from 'vue'
import TransferDatasetsDialog from './TransferDatasetsDialog.vue'
import router from '../../router'

describe('TransferDatasetsDialog', () => {
  const mockDatasets = [
    { id: '2018-06-28_13h23m10s', name: 'Untreated_3_434', uploadDT: '2018-06-28T13:23:10.837000' },
    { id: '2018-06-28_13h21m36s', name: 'Dataset 2', uploadDT: '2018-06-28T13:21:36.973867' },
    { id: '2018-06-28_12h03m36s', name: 'Dataset 3', uploadDT: '2018-06-28T12:03:36.409743' },
    { id: '2018-06-28_13h20m44s', name: 'Dataset 4', uploadDT: '2018-06-28T13:20:44.736472' },
  ]
  const mockProps = {
    currentUserId: 'current user id',
    groupName: 'Group Name',
    isInvited: true,
  }
  const stubs: Stubs = {
    DatasetItem: true,
  };
  [false, true].forEach(hasDatasets => {
    [false, true].forEach(isInvited => {
      it('should match snapshot ('
        + `${hasDatasets ? 'datasets to import' : 'no datasets'}, `
        + `${isInvited ? 'invited' : 'requesting access'})`, () => {
        const propsData = { ...mockProps, isInvited }
        const wrapper = mount(TransferDatasetsDialog, { router, propsData, stubs })
        wrapper.setData({ allDatasets: hasDatasets ? mockDatasets : [] })

        expect(wrapper).toMatchSnapshot()
      })
    })
  })

  it('should call back on success when some datasets are selected', async() => {
    const wrapper = mount(TransferDatasetsDialog, { router, propsData: mockProps, stubs })
    wrapper.setData({ allDatasets: mockDatasets })
    await Vue.nextTick()

    wrapper.findComponent(Checkbox).trigger('click')
    wrapper.findAllComponents<Button>(Button)
      .filter((b: Wrapper<Button>) => b.props().type === 'primary')
      .at(0)
      .trigger('click')
    await Vue.nextTick()

    expect(wrapper.emitted('accept')).toEqual([
      [mockDatasets.slice(1).map(ds => ds.id)],
    ])
  })
})
