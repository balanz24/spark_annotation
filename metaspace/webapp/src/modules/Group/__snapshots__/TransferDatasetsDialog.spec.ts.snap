// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransferDatasetsDialog should match snapshot (datasets to import, invited) 1`] = `
<mock-el-dialog visible="" append-to-body="" title="Join group" class="dialog" currentuserid="current user id">
  <div mock-v-loading="0">
    <div>
      <p style="margin-top: 0px;">
        Are you sure you want to join Group Name?
      </p>
      <div class="button-bar"><button type="button" class="el-button el-button--default el-button--small">
          <!---->
          <!----><span>
          Cancel
        </span></button> <button type="button" class="el-button el-button--primary el-button--small">
          <!---->
          <!----><span>
          Join group
        </span></button></div>
    </div>
  </div>
</mock-el-dialog>
`;

exports[`TransferDatasetsDialog should match snapshot (datasets to import, requesting access) 1`] = `
<mock-el-dialog visible="" append-to-body="" title="Join group" class="dialog" currentuserid="current user id">
  <div mock-v-loading="0">
    <div>
      <p style="margin-top: 0px;">
        An email will be sent to the group's principal investigator to confirm your access.
      </p>
      <div class="button-bar"><button type="button" class="el-button el-button--default el-button--small">
          <!---->
          <!----><span>
          Cancel
        </span></button> <button type="button" class="el-button el-button--primary el-button--small">
          <!---->
          <!----><span>
          Request access
        </span></button></div>
    </div>
  </div>
</mock-el-dialog>
`;

exports[`TransferDatasetsDialog should match snapshot (no datasets, invited) 1`] = `
<mock-el-dialog visible="" append-to-body="" title="Join group" class="dialog" currentuserid="current user id">
  <div mock-v-loading="0">
    <div>
      <p style="margin-top: 0px;">
        Are you sure you want to join Group Name?
      </p>
      <div class="button-bar"><button type="button" class="el-button el-button--default el-button--small">
          <!---->
          <!----><span>
          Cancel
        </span></button> <button type="button" class="el-button el-button--primary el-button--small">
          <!---->
          <!----><span>
          Join group
        </span></button></div>
    </div>
  </div>
</mock-el-dialog>
`;

exports[`TransferDatasetsDialog should match snapshot (no datasets, requesting access) 1`] = `
<mock-el-dialog visible="" append-to-body="" title="Join group" class="dialog" currentuserid="current user id">
  <div mock-v-loading="0">
    <div>
      <p style="margin-top: 0px;">
        An email will be sent to the group's principal investigator to confirm your access.
      </p>
      <div class="button-bar"><button type="button" class="el-button el-button--default el-button--small">
          <!---->
          <!----><span>
          Cancel
        </span></button> <button type="button" class="el-button el-button--primary el-button--small">
          <!---->
          <!----><span>
          Request access
        </span></button></div>
    </div>
  </div>
</mock-el-dialog>
`;
